src/Aspire.Hosting/Dashboard/TransportOptionsValidator.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal class TransportOptionsValidator(IConfiguration configuration, DistributedApplicationExecutionContext executionContext, DistributedApplicationOptions distributedApplicationOptions) : IValidateOptions<TransportOptions>
+{
+    public ValidateOptionsResult Validate(string? name, TransportOptions transportOptions)
+    {
+        var effectiveAllowUnsecureTransport = transportOptions.AllowUnsecureTransport || distributedApplicationOptions.DisableDashboard || distributedApplicationOptions.AllowUnsecuredTransport;
+
+        if (executionContext.IsPublishMode || effectiveAllowUnsecureTransport)
+        {
+            return ValidateOptionsResult.Success;
+        }
+
+        var applicationUrls = configuration[KnownConfigNames.AspNetCoreUrls];
+        if (string.IsNullOrEmpty(applicationUrls))
+        {
+            return ValidateOptionsResult.Fail($"AppHost does not have applicationUrl in launch profile, or {KnownConfigNames.AspNetCoreUrls} environment variable set.");
+        }
+
+        var firstApplicationUrl = applicationUrls.Split(";").First();
+
+        if (!Uri.TryCreate(firstApplicationUrl, UriKind.Absolute, out var parsedFirstApplicationUrl))
+        {
+            return ValidateOptionsResult.Fail($"The 'applicationUrl' setting of the launch profile has value '{firstApplicationUrl}' which could not be parsed as a URI.");
+        }
+
+        if (parsedFirstApplicationUrl.Scheme == "http" && !effectiveAllowUnsecureTransport)
```

Comment:
We need more checks here:

DOTNET_DASHBOARD_OTLP_ENDPOINT_URL

DOTNET_RESOURCE_SERVICE_ENDPOINT_URL

These endpoints are being set today by our template. These should also be checked for http. They only support a single address so that simplifies things.